docs/modules/angular/pages/cookbook/service/portal-dialog-service/index.adoc

```diff
@@ -0,0 +1,214 @@
+= OneCX PortalDialogService
+
+:dynamic_dialog_config_url: https://primeng.org/dynamicdialog#api.dynamicdialog-config
+
+== Overview
+This document aims to showcase how to manage dialogs using OneCX libraries.
+
+== Importing PortalDialogService
+To properly use PortalDialogService, please import it via the `providePortalDialogService`.
+
+.Provide PortalDialogService via the providers array
+[source,javascript]
+----
+import {
+  providePortalDialogService
+} from '@onecx/portal-integration-angular';
+
+...
+providers: [
+    providePortalDialogService(),
+]
+...
+----
+
+.Inject PortalDialogService via constructor
+[source,javascript]
+----
+import {
+  PortalDialogService
+} from '@onecx/portal-integration-angular';
+
+export class MyClass() {
+    constructor() {
+        private portalDialogService: PortalDialogService
+    }
+}
+----
+
+
+== Basic Usage
+To open the dialog use the `openDialog` method of the PortalDialogService. It returns an `Observable`, so make sure that the *call is subscribed to*.
+
+.openDialog method arguments
+[cols="3,2,5"]
+|===
+|Property name |Type | Description
+
+|`title` |`TranslationKey \| null` | Translation key for dialog title.
+|`componentOrMessage` |`Type<any> \| Type<DialogResult<T>> \| Component<T> \| TranslationKey \| DialogMessage` | Either a component or a translation key of a message with optional parameters and icon to be displayed next to the message.
+|`primaryButtonTranslationKeyOrDetails` |`TranslationKey \| ButtonDialogButtonDetails` | Translation key with optional parameters and icon to be displayed next to the text of the button.
+|`secondaryButtonTranslationKeyOrDetails?` |`TranslationKey \| ButtonDialogButtonDetails` | Translation key with optional parameters and icon to be displayed next to the text of the button.
+|`extras?` |`PortalDialogConfig` | Configuration object allowing for customization of the dialog behavior and visual aspects.
+|===
+
+.Display dialog with message
+[source,javascript]
+----
+this.portalDialogService.openDialog('TITLE_KEY', 'MESSAGE_KEY', 'PRIMARY_BUTTON_KEY', 'SECONDARY_BUTTON_KEY').subscribe((stateOnClose) => {
```

Comment:
Should we mention subscription cleanup here, or is this irrelevant/somehow handled automatically?

Reply:
I think its crucial to include the part that you actually need to subscribe, but cleanup is not related to the dialog itself working correctly I think. Dev's should know to cleanup subs from my pov